src/puppetlabs/kitchensink/file.clj

```diff
@@ -0,0 +1,84 @@
+(ns puppetlabs.kitchensink.file
+  (:import [java.io BufferedWriter FileOutputStream OutputStreamWriter]
+           [java.nio.file CopyOption Files LinkOption Paths StandardCopyOption]
+           [java.nio.file.attribute FileAttribute PosixFilePermissions]))
+
+(defn str->path
+  [path]
+  (Paths/get path (into-array String [])))
+
+(defn get-perms
+  "Returns the currently set permissions of the given file path."
+  [path]
+  (-> (str->path path)
+      (Files/getPosixFilePermissions (into-array LinkOption []))
+      PosixFilePermissions/toString))
+
+(defn set-perms
+  "Set the provided permissions on the given path. The permissions string is in
+  the form of the standard 9 character posix format, ie \"rwxr-xr-x\"."
+  [path permissions]
+  (-> (str->path path)
+      (Files/setPosixFilePermissions
+       (PosixFilePermissions/fromString permissions))
```

Comment:
should this be indented one character more? it was hard to tell at first that this was an arg to the above function.

Reply:
Some style guides say one space if there are no args on the same line as the function name. My IDE disagrees, but a lot of our projects do this.

Reply:
it's because a function call is just a list of symbols